### PR TITLE
[Core] Replace Variable template with VariableData

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
This helps two ways:
- Avoids instantion of two templates (more compilation time)
- More consistent with VariableList (which holds VariableData)

**Changelog**
- Replace Variable template with VariableData
- Added simple Parallel loop which was missing

### CODE REVIEW

**Comment by loumalouomega** [comment]
@KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
> @KratosMultiphysics/technical-committee

Ping

File: kratos/sources/model_part.cpp [hunk 1]
```diff
@@ -485,8 +486,11 @@ void ModelPart::SetNodalSolutionStepVariablesList()
         << Name() << " please call the one of the root model part: "
         << GetRootModelPart().Name() << std::endl;
 
-    for (NodeIterator i_node = NodesBegin(); i_node != NodesEnd(); ++i_node)
-        i_node->SetSolutionStepVariablesList(mpVariablesList);
+    // Iterate over nodes
+    auto& r_nodes_array = this->Nodes();
+    block_for_each(r_nodes_array,[&](NodeType& rNode) {
+        rNode.SetSolutionStepVariablesList(mpVariablesList);;
```
Comment (rubenzorrilla) [inline kratos/sources/model_part.cpp]:
```suggestion
        rNode.SetSolutionStepVariablesList(mpVariablesList);
```

**Review by rubenzorrilla — COMMENTED** [review]
Not sure if this was intentionally done to enforce passing a variable type @pooyan-dadvand. Besides this, shouldn't we change the Python export accordingly if we accept this?

**Comment by loumalouomega** [comment]
> Not sure if this was intentionally done to enforce passing a variable type @pooyan-dadvand. Besides this, shouldn't we change the Python export accordingly if we accept this?

The python it still works as we are passing a derived class, and not using anything specific from the derived class, otherwise the CI will not pass

**Comment by rubenzorrilla** [comment]
> > Not sure if this was intentionally done to enforce passing a variable type @pooyan-dadvand. Besides this, shouldn't we change the Python export accordingly if we accept this?
> 
> The python it still works as we are passing a derived class, and not using anything specific from the derived class, otherwise the CI will not pass

I know it should work, but my point is that in should be consistent no?

**Comment by loumalouomega** [comment]
> > > Not sure if this was intentionally done to enforce passing a variable type @pooyan-dadvand. Besides this, shouldn't we change the Python export accordingly if we accept this?
> > 
> > 
> > The python it still works as we are passing a derived class, and not using anything specific from the derived class, otherwise the CI will not pass
> 
> I know it should work, but my point is that in should be consistent no?

Okay

**Comment by rubenzorrilla** [comment]
Anyway, I'd wait for @pooyan-dadvand 's opinion on this before doing anything further.

**Comment by loumalouomega** [comment]
> Anyway, I'd wait for @pooyan-dadvand 's opinion on this before doing anything further.

Done anyway

**Comment by loumalouomega** [comment]
> Anyway, I'd wait for @pooyan-dadvand 's opinion on this before doing anything further.

Looks like due to pybind11 limitations, we must do the export as it is now

**Comment by pooyan-dadvand** [comment]
I believe that it comes from the Kratos v0 and can be safely changed to the VariableData.

**Review by rubenzorrilla — APPROVED** [review]
I think that now we can safely merge this. About the Python export, we can leave it for the future. Thanks for the patience @loumalouomega

**Pull request merged**